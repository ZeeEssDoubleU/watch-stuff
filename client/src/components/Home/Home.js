import React, { Component } from "react";

import VideoGrid from "../VideoGrid/VideoGrid.js";
import Sidebar from '../SideBar/SideBar.js';
import './Home.scss';

class Home extends Component {
	render() {
		return (
			<>
				<Sidebar />
				<div className="home">
					<div className="responsive-video-grid-container">
						<VideoGrid title="Trending" />
						<VideoGrid title="Autos & Vehicles" hideDivider={true} />
					</div>
				</div>
			</>
		);
	}
}

export default Home;
