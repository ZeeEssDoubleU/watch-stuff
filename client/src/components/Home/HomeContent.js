import React, { useState, useEffect } from "react";
import { connect } from "react-redux";

import "./HomeContent.scss";
import VideoGrid from "../VideoGrid/VideoGrid";
import InfiniteScroll from "../InfiniteScroll/InfiniteScroll";

import {
	selector_mostPopularVideos,
	selector_mostPopularVideosByCategory,
	selector_mostPopularVideosByCategoryLoaded,
	selector_mostPopularVideosByCategoryLength,
} from "../../store/reducers/videos";

const HomeContent = props => {
	const [lazyLoadIndex, setLazyLoadIndex] = useState(0);

	useEffect(() => {
		const homeContentHeight =
			document.querySelector(".home-content").offsetHeight -
			document.querySelector(".loader-container").offsetHeight;
		if (
			// if height of loaded content is less than window height, load more content
			props.videosByCategoryLoaded &&
			homeContentHeight < window.innerHeight
		) {
			lazyLoadVideoCategories();
		}
	});

	// function that loads data to page.  Data already fetched from server
	const lazyLoadVideoCategories = () => {
		if (
			props.videosByCategoryLoaded &&
			lazyLoadIndex < props.videosByCategoryLength
		) {
			setLazyLoadIndex(lazyLoadIndex + 1);
		}
	};

	const shouldShowLoader = () => {
		// if vidoesByCategory loaded, return true if lazyLoadIndex < total categories
		return props.videosByCategoryLoaded &&
			lazyLoadIndex < props.videosByCategoryLength
			? true
			: false;
	};

	const categoryNames = Object.keys(props.videosByCategory);
	const categoryGrids = categoryNames
		.slice(0, lazyLoadIndex)
		.map((categoryName, index) => (
			<VideoGrid
				title={categoryName}
				key={categoryName}
				videos={props.videosByCategory[categoryName]}
				hideDivider={index === categoryNames.length - 1}
			/>
		));

	return (
		<div className="home-content">
			<div className="responsive-video-grid-container">
				<InfiniteScroll
					bottomReachedCallback={lazyLoadVideoCategories}
					showLoader={shouldShowLoader()}>
					<VideoGrid title="Trending" videos={props.mostPopular} />
					{categoryGrids}
				</InfiniteScroll>
			</div>
		</div>
	);
};

const mapStateToProps = state => ({
	mostPopular: selector_mostPopularVideos(state),
	videosByCategory: selector_mostPopularVideosByCategory(state),
	videosByCategoryLoaded: selector_mostPopularVideosByCategoryLoaded(state),
	videosByCategoryLength: selector_mostPopularVideosByCategoryLength(state),
});

export default connect(
	mapStateToProps,
	null,
)(HomeContent);
