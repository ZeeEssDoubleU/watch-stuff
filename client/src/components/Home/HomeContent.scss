@import "../../styles/shared.scss";

.home-content {
   margin-top: $content-margin-top;
   margin-left: $content-margin-left;
   margin-right: $content-margin-right;

   display: grid;
   grid: auto / auto;
   justify-content: center;
   align-items: center;
   .responsive-video-grid-container {
      margin: auto;
      max-width: 1284px;
   }
}