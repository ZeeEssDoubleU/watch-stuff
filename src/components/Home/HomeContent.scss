@import "../../styles/shared.scss";

.home-content {
   display: grid;
   grid: auto / auto;
   justify-content: center;
   padding: 24px;

   .responsive-content {
      max-width: 1284px;
      margin: auto;
   }
}
