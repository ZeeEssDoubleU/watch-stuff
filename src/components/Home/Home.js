import React, { Component} from 'react';

import VideoPreview from '../VideoPreview/VideoPreview.js';
import './Home.scss';

class Home extends Component {
   render() {
      return (
         <div className="home">
            <VideoPreview />
         </div>
      );
   }
}

export default Home;