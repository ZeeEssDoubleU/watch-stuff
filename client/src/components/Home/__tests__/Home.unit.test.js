import React from "react";
import { shallow } from "enzyme";

import Home from "../Home";

test("renders Home", () => {
   const wrapper = shallow(<Home />);
   expect(wrapper).toMatchSnapshot();
});
