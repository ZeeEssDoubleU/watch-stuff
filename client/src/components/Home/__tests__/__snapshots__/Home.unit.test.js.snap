// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home 1`] = `
<Fragment>
  <SideBar />
  <div
    className="home"
  >
    <div
      className="responsive-video-grid-container"
    >
      <VideoGrid
        title="Trending"
      />
      <VideoGrid
        hideDivider={true}
        title="Autos & Vehicles"
      />
    </div>
  </div>
</Fragment>
`;
